### ISSUE TITLE
Disallow r.minval and r.maxval as secondary index functions

### ISSUE THREAD

**Issue by danielmewes** [rethinkdb/rethinkdb#5222]
`r.minval` and `r.maxval` can currently become a secondary index key, if you return them from a secondary index function.

We think that this might break some code and should probably disallow those values to be stored as secondary index keys.

@Tryneus I think you were using this to emulate "dense" indexes right? Did you just need a way to store something equivalent to `null` in an index? Or also a special value to represent "field missing"?

**Comment by Tryneus** [comment]
@danielmewes - yes, this was to implement 'dense' indexes.  The index creating function I'm using is:

```js
r.table(this.name).indexCreate(index_id, (row) =>
  r.expr(fields).map((field_name) => row(field_name).default(r.minval)))
```

If we could create indexes with a flag such that this works without `r.minval`, I would be ok with that.  It would be nice to be able to differentiate between `field == null` and `field is missing`.

**Comment by danielmewes** [comment]
Fix in review 3535

**Comment by danielmewes** [comment]
Fixed in next by a57b599
